llvm/lib/TargetParser/Triple.cpp

```diff
@@ -1510,6 +1510,8 @@ VersionTuple Triple::getDXILVersion() const {
   if (getArch() != dxil || getOS() != ShaderModel)
     llvm_unreachable("invalid DXIL triple");
   StringRef Arch = getArchName();
+  if (Arch == "dxil")
```

Comment:
I think you can say `getSubArch() == NoSubArch` here instead of using string comparison

Reply:
Fixed.